tests/ui/asm/naked-asm-outside-naked-fn.rs

```diff
@@ -12,24 +12,24 @@ fn main() {
     test1();
 }
 
-#[naked]
+#[unsafe(naked)]
 extern "C" fn test1() {
-    unsafe { naked_asm!("") }
+    naked_asm!("")
 }
 
 extern "C" fn test2() {
-    unsafe { naked_asm!("") }
+    naked_asm!("")
     //~^ ERROR the `naked_asm!` macro can only be used in functions marked with `#[naked]`
```

Comment:
Perhaps we want to update this message to say "...marked with `#[unsafe(naked)]`" so that people who do what the error message implies don't immediately get another error.

Reply:
Ah, I see @tgross35 concurrently and coincidentally mentioned this as well in https://github.com/rust-lang/rust/pull/139753#issuecomment-2816218717.

Reply:
Done. Should we use `#[unsafe(naked)]` everywhere in comments, docs, ect? (I can do that in the stabilization PR, so we can move forward here)

Reply:
I mean, the attribute name is still just `naked`.  Really where I feel compelled to do it is anywhere that it's going to prompt the user to add it to the code.  Other cases would have to be looked at for what makes sense.  You might also look at what we did with `no_mangle` and friends (though no guarantees whatever was done there was perfectly what we want).